ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -12975,8 +12982,16 @@ void HqlCppTranslator::doBuildAggregateProcessTransform(BuildCtx & ctx, BoundRow
             }
             break;
         }
+
+        if (isAggregateOp && isDynamicOffset)
+            throwError1(HQLERR_AggregateDyanmicOffset, str(target->queryChild(1)->queryId()));
+
         if (target->queryType()->getSize() == UNKNOWN_LENGTH)
+        {
            isVariableOffset = true;
+           if (isAggregateOp)
+               isDynamicOffset = true;
```

Comment:
The above logic (2 IF stmts) looks odd, may either be wrong or could do with some comments.
isAgg. is reset inside the loop so if (isAggregateOp && isDynamicOffset) can only be hit if isDynamicOffset was changed to true on a previous loop (not reset on each iteration) and the current target (iterated over) is still an aggregaetOp. Why does there need to be more than one aggOp encountered before throwing (when encountering UNKNOWN_LENGTH)? Looks like it should just be a single IF?